Add pointer generator validation

Wendy-Xiao opened the pull request:
Make new validate_ptr
Comment out visual features and CNN in train_ptr
Make new sample function in decoder in model_ptr
TODO: beam search of sample in model

Wendy-Xiao committed:
Add pointer generator validation

Make new validate_ptr
Comment out visual features and CNN in train_ptr
Make new sample function in decoder in model_ptr
TODO: beam search of sample in model

TamakiSakura commented on train_ptr.py:
```diff
@@ -74,14 +74,15 @@ def main(args):
-	     list(encoder.linear.parameters()) + list(encoder.bn.parameters())
+    params = list(layout_encoder.parameters()) + list(decoder.parameters())
+    # params = list(layout_encoder.parameters()) + list(decoder.parameters()) + \
+	   #   list(encoder.linear.parameters()) + list(encoder.bn.parameters())
```
You don't need these changes

Wendy-Xiao replied:
Just to make it convenient when there is no visual feature, in that case, we have no encoder. You can keep it or delete it, doesn't matter.

TamakiSakura commented on model.py:
```diff
@@ -136,3 +136,25 @@ def sample(self, features, states=None):
+            sampled_ids.append(predicted)
+
+        sampled_ids = torch.cat(sampled_ids, 1)                  # (batch_size, 20)
+        return sampled_ids
```
Was this version working or you're just keeping it here because it's more convenient?
I am fine with keep working copy in repo but I would like to know if I need to review it seriously.

Wendy-Xiao replied:
No need to review it seriously, it's just a draft now. I may finish it after the presentation.

Wendy-Xiao committed:
slight changes in sample

Move the one hot variables out of the loop, for purpose of saving time and space.
If cuda is available, make mask in cuda.

Wendy-Xiao committed:
Add comments and simplified hidden
